Extra unit tests and data loader integration tests

pieterlukasse opened the pull request:
# What? Why?
This PR includes more checks in the `TestIntegrationTest.java` increasing its coverage of the different data types supported by the data model (continuing what was started in #1399). It also includes more unit tests for clinical data loading and mRNA expression data loading. 

This PR also includes some small fixes in two other test classes to try to prevent failures due to execution order of the unit tests. 

Code and/or comments clean-up was done where applicable in `ImportClinicalData.java`. Because we now make test assertions on warnings given by the loader, it was important to add the option to reset the warnings list in `ProgressMonitor.java`

Changes proposed in this pull request:
- see commit list


# Notify reviewers
@ersinciftci @morungos @ecerami @sheridancbio

jjgao commented:
@sheridancbio this is a continuation of #1399. Would you be able to review this as well?

sheridancbio commented on core/src/test/java/org/mskcc/cbio/portal/scripts/TestIntegrationTest.java:
```diff
@@ -176,7 +205,76 @@ public void testLoadStudyEs0() {
+                    gisticChr10 = gistic;
+                }
+            }
+            assertEquals(8, gisticChr10.getGenes_in_ROI().size());
```
ROI is a non-obvious acronym to a new developer. Maybe change this function to spell out what this means (RegionOfInterest?)  Also, google java style specifies that method names should be lower camel case : https://google.github.io/styleguide/javaguide.html#s5.2.3-method-names. Refactoring this may be outside the scope of this PR .. but maybe it can be done without too much pain.

pieterlukasse replied:
I agree. Maybe a new PR can take care of this. Anyway, `Gistic.java` does specify what ROI means, so that is already of some help.

pieterlukasse replied:
or maybe one of the new API development efforts will take case of this anyway...

sheridancbio commented on core/src/test/java/org/mskcc/cbio/portal/scripts/TestIntegrationTest.java:
```diff
@@ -176,7 +205,76 @@ public void testLoadStudyEs0() {
+            for (Serializable profileEntry: methylationProfileData) {
+                String profileData = ((DBSimpleProfileData)profileEntry).profile_data;
+                double profileDataValue = Double.parseDouble(profileData);
+                if (profileDataValue >= 0.5 && profileDataValue < 0.6) {
```
Same range used here (for methylation) as for CNA profiles. This is surprising, but I don't know these data values.

pieterlukasse replied:
same trick

sheridancbio replied:
ok

sheridancbio commented on core/src/test/java/org/mskcc/cbio/portal/servlet/TestWebService.java:
```diff
@@ -89,6 +89,8 @@ public void setUp() throws DaoException {
       expect(control.getUserDetails()).andStubReturn(mockUserDetails);
       replay(control);
       SpringUtil.setAccessControl(control);
+      
```
drop blank line

sheridancbio commented on core/src/test/resources/clinical_data_small.txt:
```diff
@@ -13,5 +13,8 @@ TCGA-BH-A1F0	TCGA-BH-A1F0-01	basal-like	DECEASED	25.79	Recurred/Progressed	NA
 TCGA-B6-A0I6	TCGA-B6-A0I6-01	NA	DECEASED	32.56	Recurred/Progressed	NA
 TCGA-BH-A18V	TCGA-BH-A18V-01	basal-like	DECEASED	51.09	NA	NA
 TCGA-BH-A18Q	TCGA-BH-A18Q-01	basal-like	DECEASED	55.59	NA	NA
+abc
```
I suppose this is to generate a data format warning / exception on import?

pieterlukasse replied:
yes, on purpose

sheridancbio commented on core/src/test/resources/clinical_data_small.txt:
```diff
@@ -13,5 +13,8 @@ TCGA-BH-A1F0	TCGA-BH-A1F0-01	basal-like	DECEASED	25.79	Recurred/Progressed	NA
 TCGA-BH-A18K	TCGA-BH-A18K-01	basal-like	DECEASED	90.74	Recurred/Progressed	NA
-TCGA-BH-A18K	TCGA-BH-A18K-01	basal-like	DECEASED	90.74	Recurred/Progressed	NA
\ No newline at end of file
+
```
I suppose this is to generate a "blank line" warning on import?

pieterlukasse replied:
yes, this is on purpose

pieterlukasse commented on core/src/test/java/org/mskcc/cbio/portal/scripts/TestIntegrationTest.java:
```diff
@@ -149,6 +161,23 @@ public void testLoadStudyEs0() {
+            for (Serializable profileEntry: cnaProfileData) {
+                String profileData = ((DBSimpleProfileData)profileEntry).profile_data;
+                double profileDataValue = Double.parseDouble(profileData);
+                if (profileDataValue > -0.6 && profileDataValue <= -0.5) {
```
for CNA data we have discrete and continuous values ("log2CNA"). These last ones are used in various plots, such as the mutations vs CNA plot in the study view. 

The if statement here is on purpose, as I am counting the number of values in this loop that are in the range 0.5 to 0.6. I manually verified in the file that this should be 273 values, and this is asserted here in this test (row 180). 

I will add a comment to this block to make this a bit more clear.

sheridancbio replied:
ok .. understood

sheridancbio commented:
Still waiting for squash and rebase ... then will merge.

pieterlukasse commented on core/pom.xml:
```diff
@@ -437,5 +437,36 @@
+                                <pluginExecutionFilter>
+                                    <groupId>org.codehaus.mojo</groupId>
+                                    <artifactId>
+                                        sql-maven-plugin
```
done

pieterlukasse commented on core/pom.xml:
```diff
@@ -437,5 +437,36 @@
+                                        <goal>execute</goal>
+                                    </goals>
+                                </pluginExecutionFilter>
+                                <action>
```
I get an error in Eclipse if I do so...
![image](https://cloud.githubusercontent.com/assets/2900303/18663615/87fafc5e-7f1f-11e6-83e7-108d68315e13.png)

sheridancbio replied:
Ok, then leave them in. They seemed semantically empty to me, but maybe they specify that the action to take is a "null action" / no-op

pieterlukasse commented on core/src/main/java/org/mskcc/cbio/portal/scripts/ImportClinicalData.java:
```diff
@@ -251,13 +274,13 @@ private void importData(BufferedReader buff, List<ClinicalAttribute> columnAttrs
             }
 
-            String[] fields = getFields(line, columnAttrs);
+            String[] fields = getFields(line, untrimmedLine, columnAttrs);
```
yes, this is odd. I will rewrite it.

pieterlukasse commented on core/src/main/java/org/mskcc/cbio/portal/util/ProgressMonitor.java:
```diff
@@ -246,6 +246,14 @@ public static void logDebug(String debugMessage) {
+     * Reset the warnings list.
+     */
+    public static void resetWarnings() {
+    	progressMonitor.warnings = new TreeSet<>();
```
done

pieterlukasse commented on core/src/test/java/org/mskcc/cbio/portal/scripts/TestImportClinicalData.java:
```diff
@@ -68,13 +69,23 @@
-		DaoGeneticProfile.reCache();
-		DaoPatient.reCache();
-		DaoSample.reCache();
+        DaoCancerStudy.reCacheAll();
```
yeah...I thought my Eclipse was using the `.editorconfig` file that is part of this project (see https://github.com/cBioPortal/cbioportal/pull/593)...but apparently something is still wrong. Anyway, I made a small fix just for this part.

pieterlukasse commented on core/src/test/java/org/mskcc/cbio/portal/scripts/TestImportClinicalData.java:
```diff
@@ -109,50 +118,88 @@ public void testImportClinicalDataNewStudy() throws Exception {
         caseSet.add("TCGA-A1-A0SE");
         caseSet.add("TCGA-A1-A0SI");
-
+        
```
I love blank lines separating parts :smile:

pieterlukasse commented on core/src/test/java/org/mskcc/cbio/portal/scripts/TestImportClinicalData.java:
```diff
@@ -185,28 +232,53 @@ else if (patientData.getStableId().equals("TCGA-A1-A0SE")) {
+
+        checkDataSlice(cancerStudy);
+	}
+	
```
ok, removed consecutive ones

pieterlukasse commented on core/src/test/java/org/mskcc/cbio/portal/scripts/TestImportTabDelimData.java:
```diff
@@ -332,20 +338,43 @@ public void testImportmRnaData2() throws Exception {
+		int countInvalidEntrez = 0;
+		int countSkippedWarnings = 0;
+		for (String warning: warnings) {
+		    if (warning.contains("Duplicated row"))
```
fixed

pieterlukasse commented on core/src/test/java/org/mskcc/cbio/portal/scripts/TestIntegrationTest.java:
```diff
@@ -149,6 +161,23 @@ public void testLoadStudyEs0() {
+            geneticProfileStableIds = new ArrayList<String>();
+            geneticProfileStableIds.add("study_es_0_log2CNA");
+            hugoGeneSymbols = new ArrayList<String>(Arrays.asList("ACAP3","AGRN","ATAD3A","ATAD3B","ATAD3C","AURKAIP1","ERCC5"));
+    		cnaProfileData = apiService.getGeneticProfileData(geneticProfileStableIds, hugoGeneSymbols, null, null);
```
done

pieterlukasse commented:
hi @sheridancbio thanks for the review! I addressed most of your comments. You can find this in the new commits. Let me know if this is OK, then I will squash them.

sheridancbio commented:
One comment on the latest changes: in ImportClinicalData.java we will now be trimming every field parsed from the table. I think this is probably the correct thing to do, but is there any chance some of the inner fields *should* be allowed to have leading whitespace? I doubt it, but just want to raise the question. Otherwise, I let's go forward with squashing and rebasing to the front of rc.

pieterlukasse commented:
@sheridancbio no, I cannot think of any scenario where leading or trailing white spaces would be necessary in one of the clinical attributes. I think however that it is good for @zheins to test this with his data to make sure it runs smoothly on your side as well. Shall we wait for that?

sheridancbio commented:
I will ask @zheins about it when he gets in ... if he cannot think of cases needing leading spaces, I'll merge. Also, I see that there is an option for me to squash the commits during the merge. I'll use that option ... unless you want to squash in your own way (The auto-squash gives a single changeset as the result)

pieterlukasse commented:
@sheridancbio I squashed just now in the best way I could, trying to keep "topical commits".

sheridancbio commented:
@zheins and @ritikakundra said that we don't want to allow or support the import of flanking whitespace in fields during Clinical Data import.

sheridancbio commented:
I'm ready to merge, but I think 6 changesets is still too many for the scope of this work. Can you please also squash the following commits: 
6451eda05ccaf0de332cbf26b6ab8a593ac52d6a
f9216e36acf5ebbb6c9ba596d189e162ff2a56a8
ce974159643dd4aae25e3d466eca3ddadc4376d5
That will leave three remaining commits to persist on the cBioPortal rc branch history
